### PR TITLE
suggestions

### PR DESCRIPTION
**Pull request by manuartero**
Hi Cecilia,
great job on solving this exercise!

I can see the effort you put in, and I'm glad you were able to solve it. 
Watching the video, I'm 100% sure you grasped the basics. For a professor, acknowledging that a student is learning is a great feeling.

I think you did pretty a good job and **you can be proud of this step.**

***

I took the liberty of cloning your repo. Honestly last week i'd have done a deeper review but i'm afraid I have little time now. 
anyway you asked two things at the end of your video and i can solve those :)

hope you find this useful.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "prop-types": "^15.8.1",
```
Comment (manuartero) [inline package.json:13]: in pure js , the linter is expecting you to use this "props - validation - library"

This validation is demanded by the linter only for js projects because  if the project were **Typescript**, this props validation is already done by Ts itself...

2 options: 

1. supress the linter warning (NOT recommended)
2. install (`npm i prop-types`) and call the library as following

File: src/assets/components/InputBox.jsx [hunk 2]
```diff
@@ -11,16 +11,20 @@ function InputBox(props) {
...
             <input onChange={handleChange} type="text" value={inputText}/>
             <button onClick={() => {
                 props.click(inputText);
                 setInputText("")}}>
                 <span>Add</span>
             </button>
         </div>
-    )   
-    
+    )
```
Comment (manuartero) [inline src/assets/components/InputBox.jsx:22]: just prettier format on save, please ignore

File: src/assets/components/InputBox.jsx [hunk 3]
```diff
@@ -11,16 +11,20 @@ function InputBox(props) {
...
+
 }
 
-export default InputBox;
\ No newline at end of file
+InputBox.propTypes = {
+    click: PropTypes.func.isRequired,
+};
+
```
Comment (manuartero) [inline src/assets/components/InputBox.jsx:29]: props validation

File: src/assets/components/ListItem.css [hunk 4]
```diff
@@ -1,11 +1,5 @@
-/* .listItem {
-  text-align: left;
-  position: relative;
-  padding: 2rem;
-  font-size: 2rem;
-  color:  #FF71CD
-} */
```
Comment (manuartero) [inline src/assets/components/ListItem.css:7]: leftover

File: src/assets/components/List.jsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import "./List.css";
 import InputBox from "./InputBox";
 import ListItem from "./ListItem";
-import React, { useState } from "react";
+import { useState } from "react";
```
Comment (manuartero) [inline src/assets/components/List.jsx:4]: as you noticed in the video, `React` isn't required. the only import you need is:

```js
import { /* here the hooks you use */ } from 'react'
```

why do code generators (like chatGPT, or Copilot) include then this:

```js
import React from 'react'
```

well this was needed in prev. versions of React (I think React < v17), all the code generated before that version does include this import statement, that's why the generators include it. 

it was needed so the transpiler did know you were going to write some JSX syntax (if i recall correctly) but not needed anymore

TL;DR you can remove that safely

File: src/assets/components/ListItem.jsx [hunk 6]
```diff
@@ -6,40 +8,49 @@ function ListItem(props) {
...
\ No newline at end of file
+ListItem.propTypes = {
+  id: PropTypes.number.isRequired,
+  text: PropTypes.string.isRequired,
+  onChecked: PropTypes.func.isRequired,
+  onDoubleClick: PropTypes.func.isRequired,
+  isEditing: PropTypes.bool.isRequired,
+  onChange: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+}
```
Comment (manuartero) [inline src/assets/components/ListItem.jsx:54]: prop validation, now you don't have the red warnings :)
